graph/patterns/viewpoint.md

```diff
@@ -0,0 +1,112 @@
+    </Annotations>
+```
+- An alternative to this design would be to store the user state on the client side. However, this may be problematic in some cases, because of the many devices that a user may have and the need to synchronize the state across them.
+
+## Examples
+
+### Defining a viewpoint
+
+The following example demonstrates how to define the 'viewpoint' property for the `chat` entity, where a chat is a collection of chatMessages between one or more participants: 
+```
+  <ComplexType Name="chatViewpoint" >
+        <Property Name="isHidden" Type="Edm.Boolean" />
+        <Property Name="lastMessageReadDateTime" Type="Edm.DateTimeOffset" />
+  </ComplexType>
+
+  <EntityType Name="chat" BaseType="graph.entity" >
+        <Property Name="chatType" Type="graph.chatType" Nullable="false" />
+        <Property Name="createdDateTime" Type="Edm.DateTimeOffset" />
+        <Property Name="lastUpdatedDateTime" Type="Edm.DateTimeOffset" />              
+        <Property Name="topic" Type="Edm.String" />
+        <Property Name="viewpoint" Type="graph.chatViewpoint" />
+       ...
+        <NavigationProperty Name="tabs" Type="Collection(graph.teamsTab)" ContainsTarget="true" />
+  </EntityType>
+
+```
+### Reading an entity with a viewpoint
+
+The following example shows reading a collection of chats for an identified user, with a viewpoint for each chat:
+
+```http
+GET https://graph.microsoft.com/v1.0/users/8b081ef6-4792-4def-b2c9-c363a1bf41d5/chats
+```
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+{
+    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#chats",
+    "@odata.count": 3,
+    "value": [
+        {
+            "id": "19:meeting_MjdhNjM4YzUtYzExZi00OTFkLTkzZTAtNTVlNmZmMDhkNGU2@thread.v2",
+            "topic": "Meeting chat sample",
+            "createdDateTime": "2020-12-08T23:53:05.801Z",
+            "lastUpdatedDateTime": "2020-12-08T23:58:32.511Z",
+            "chatType": "meeting",         
+            "viewpoint":{
+                "lastMessageReadDateTime": "2021-03-28T21:10:00.000Z" // User has unread messages
+            }
+        },
+        {
+            "id": "19:561082c0f3f847a58069deb8eb300807@thread.v2",
+            "topic": "Group chat sample",
+            "createdDateTime": "2020-12-03T19:41:07.054Z",
+            "lastUpdatedDateTime": "2020-12-08T23:53:11.012Z",
+            "chatType": "group",            
+            "viewpoint":{
+                "lastMessageReadDateTime": "0000-01-01T00:00:00.000Z" // User hasnt read anything since no message was posted
+            }
+        }
+    ]
+}
+```
+### Updating a viewpoint
+
+You can update the `viewpoint` property only if the server does not compute it automatically. Updating the `viewpoint` property usually has a side effect, so you may want to use an OData action to perform the update.
+
+The following example shows marking a chat as read for a user:
```

Comment:
We've said that a viewpoint is specific view for the signed-in user.  However, the action below seems to imply that a privileged user (like an admin) could change the state (chat read) for another user.  Is this something we really want to show as an example?

Do the guidelines permit user A updating user B's viewpoint?

Reply:
Based on the current design, the Read operation cannot return another user's viewpoint, while Update seems possible; therefore, a privileged user can potentially update viewpoints of other users. so far i can't think of a good example.

Reply:
There is no reason that an API cannot exist to change the read state for a user other than the calling user.  This example doesn't use the viewPoint to make this change.  
I am going to argue that you could not PATCH an entity and update a viewPoint for any user other than the signed in user.  However, I need to go ask how chatViewPoint has been implemented, because I also don't think you should be able to read the viewPoint for a different user.  e.g. via /users/{notsignedinuser}/chats/{chatId}

Reply:
@dkershaw10 i moved a blurb about update operations to issues and considerations.